File: 15-trees1/4-recursive-algorithms/main.java
// Recursive Tree Algorithms - Java Example

class TreeNode {
    int data;
    TreeNode left, right;
    TreeNode(int data) {
        this.data = data;
        left = right = null;
    }
}

public class RecursiveTreeAlgorithms {

    public static int treeHeight(TreeNode node) {
        if (node == null) return 0;
        int leftHeight = treeHeight(node.left);
        int rightHeight = treeHeight(node.right);
        return Math.max(leftHeight, rightHeight) + 1;
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);

        System.out.println("Tree height: " + treeHeight(root));
    }
}

File: 15-trees1/4-recursive-algorithms/main.cpp
# Recursive Tree Algorithms - Python Example

class TreeNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def tree_height(node):
    if node is None:
        return 0
    left_height = tree_height(node.left)
    right_height = tree_height(node.right)
    return max(left_height, right_height) + 1

root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)

print("Tree height:", tree_height(root))
